### PR TITLE
Clean up Terraform resource names (#2033)

### PR DESCRIPTION
**Pull request by natanlao**
Rename Terraform resources for consistency, and add a script that
renames existing resources when `make deploy` is called.

### CODE REVIEW

File: common.mk [hunk 1]
```diff
@@ -74,6 +74,10 @@ check_aws: check_python
 		false; \
 	fi
 
+.PHONY: rename_resources
+rename_resources:
+	python $(project_root)/scripts/rename_resources.py
```
Comment (jessebrennan) [inline common.mk:79]: I'm assuming that you've checked this with Hannes, but I'm a little surprised that we want a make target for this script. My understanding was that this was something that would only need to be run once for each deployment. Then the script could be archived in a later PR.

File: terraform/elasticsearch.tf.json.template.py [hunk 2]
```diff
@@ -108,6 +107,6 @@
                     }
                 }
             }
-        }
+        } if domain else {}
```
Comment (jessebrennan) [inline terraform/elasticsearch.tf.json.template.py:110]: This seems like an unrelated change. Can you explain why you added this?

Comment (jessebrennan) [reply terraform/elasticsearch.tf.json.template.py:110]: Does this solve the [0] issue with the aws_eslaticsearch_domain name?

File: scripts/rename_resources.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+if __name__ == '__main__':
+    configure_script_logging(log)
+    tf_state = subprocess.run(['terraform', 'state', 'list'], **run_args)
+    current = [r.strip() for r in tf_state.stdout.decode('utf-8').split('\n')]
+    for resource in filter(lambda r: r in renamed.keys(), current):
+        log.info('Found %r, renaming to %r', resource, renamed[resource])
+        subprocess.run(['terraform', 'state', 'mv', resource, renamed[resource]],
+                       **run_args)
```
Comment (jessebrennan) [inline scripts/rename_resources.py:52]:
```suggestion
                       **run_args).check_returncode()
```

If not this exactly, there should be some checking of the return code here.

File: scripts/rename_resources.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+run_args = {
+    'cwd': TERRAFORM_PATH,
+    'check': True,
+    'capture_output': True,
+    'shell': False
+}
+
+if __name__ == '__main__':
+    configure_script_logging(log)
+    tf_state = subprocess.run(['terraform', 'state', 'list'], **run_args)
```
Comment (jessebrennan) [inline scripts/rename_resources.py:47]:
```suggestion
                       **run_args).check_returncode()
```

If not this exactly, there should be some checking of the return code here.

File: scripts/rename_resources.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    current = [r.strip() for r in tf_state.stdout.decode('utf-8').split('\n')]
+    for resource in filter(lambda r: r in renamed.keys(), current):
+        log.info('Found %r, renaming to %r', resource, renamed[resource])
+        subprocess.run(['terraform', 'state', 'mv', resource, renamed[resource]],
```
Comment (jessebrennan) [inline scripts/rename_resources.py:51]: What happens if you run this twice? Is `terraform state mv` idempotent?

File: common.mk (hunk 1 above)
Comment (natanlao) [reply common.mk:79]: Yeah, Hannes requested that the script be injected into the build process specifically. I'm not sure if there's a strong advantage beyond keeping UPGRADING.rst a little smaller. I'll raise this during standup.

File: terraform/elasticsearch.tf.json.template.py (hunk 2 above)
Comment (natanlao) [reply terraform/elasticsearch.tf.json.template.py:110]: Yes -- Hannes requested this in the original ticket.

File: scripts/rename_resources.py (hunk 3 above)
Comment (natanlao) [reply scripts/rename_resources.py:52]: The `'check': True` in `**run_args` should enable this functionality

File: common.mk (hunk 1 above)
Comment (jessebrennan) [reply common.mk:79]: It's ok, I was just surprised. If that's what you were instructed to do, it sounds good to me :+1:

Comment (jessebrennan) [inline common.mk:79]: I thought I already left a comment about this in my review, but I can't find it. Essentially: this target should depend on the `init` target in `terraform/Makefile`. That probably means this target needs to be moved to that Makefile.

File: scripts/rename_resources.py (hunk 5 above)
Comment (natanlao) [reply scripts/rename_resources.py:51]: `terraform state mv` is idempotent and will fail if trying to rename a nonexistent resource

File: common.mk (hunk 1 above)
Comment (natanlao) [reply common.mk:79]: @hannes-ucsc

Comment (hannes-ucsc) [reply common.mk:79]: The size of UPGRADING.rst isn't a concern but the need for manual upgrade intervention is. Implementing this as specified in the ticket means avoiding the manual intervention when these changes are deployed to any deployment. I'm happy to discuss this further, but be sure to read the ticket description first, @jessebrennan.

File: scripts/rename_resources.py (hunk 5 above)
Comment (hannes-ucsc) [reply scripts/rename_resources.py:51]: If it fails, it is NOT idempotent. Do we need to swallow the error or do we want to fail since a check above should prevent the case?

File: scripts/rename_resources.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    configure_script_logging,
+)
+
+log = logging.getLogger(__name__)
+
+TERRAFORM_PATH = 'terraform'
+try:
+    TERRAFORM_PATH = os.path.join(os.environ['project_root'], TERRAFORM_PATH)
+except KeyError:
+    log.error("Please run 'source environment' from the project root.")
```
Comment (hannes-ucsc) [inline scripts/rename_resources.py:15]: The makefile should check this. No need to duplicate this error message here.

File: scripts/rename_resources.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    'check': True,
+    'capture_output': True,
+    'shell': False
+}
+
+if __name__ == '__main__':
+    configure_script_logging(log)
+    tf_state = subprocess.run(['terraform', 'state', 'list'], **run_args)
+    current = [r.strip() for r in tf_state.stdout.decode('utf-8').split('\n')]
+    for resource in filter(lambda r: r in renamed.keys(), current):
```
Comment (hannes-ucsc) [inline scripts/rename_resources.py:49]: Factor this into a function called main() and make run_args a local var of that

File: scripts/rename_resources.py [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+log = logging.getLogger(__name__)
+
+TERRAFORM_PATH = 'terraform'
+try:
+    TERRAFORM_PATH = os.path.join(os.environ['project_root'], TERRAFORM_PATH)
+except KeyError:
+    log.error("Please run 'source environment' from the project root.")
+    exit(1)
+
+renamed = {
```
Comment (hannes-ucsc) [inline scripts/rename_resources.py:18]: this can remain global

File: scripts/rename_resources.py [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+import os.path
+import subprocess
+
+from azul.logging import (
+    configure_script_logging,
+)
+
+log = logging.getLogger(__name__)
+
+TERRAFORM_PATH = 'terraform'
```
Comment (hannes-ucsc) [inline scripts/rename_resources.py:11]: We don't use upper case for pseudo constants. COuld you add a rule to the C guide?

Besides, it's fine to inline this altogether. There is only one reference. I don't believe in trying to anticipate anything that might vary in the future and extract it into a pseudo-constant.

File: scripts/rename_resources.py (hunk 7 above)
Comment (hannes-ucsc) [inline scripts/rename_resources.py:49]: Replace filter with `if`. `for x in filter()` is a usually a bad smell.

File: scripts/rename_resources.py (hunk 9 above)
Comment (natanlao) [reply scripts/rename_resources.py:11]: > We don't use upper case for pseudo constants. COuld you add a rule to the C guide?

#2128

> Besides, it's fine to inline this altogether. There is only one reference. I don't believe in trying to anticipate anything that might vary in the future and extract it into a pseudo-constant.

I used this form to avoid a `Name 'TERRAFORM_PATH' can be undefined` warning if it is defined only inside the `try:` block. In cases like this, is there a better way to write it?

File: scripts/rename_resources.py [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+    'cwd': TERRAFORM_PATH,
+    'check': True,
+    'capture_output': True,
+    'shell': False
+}
+
+if __name__ == '__main__':
+    configure_script_logging(log)
+    tf_state = subprocess.run(['terraform', 'state', 'list'], **run_args)
+    current = [r.strip() for r in tf_state.stdout.decode('utf-8').split('\n')]
```
[the remark this answers is not in the document]
Comment (natanlao) [reply scripts/rename_resources.py:48]: I see now that if I were using `.split()`, it would have been better to write

    tf_state.stdout.split(b'\n')

but `splitlines` is better in this case.

File: scripts/rename_resources.py (hunk 5 above)
Comment (natanlao) [reply scripts/rename_resources.py:51]: Summary from standup: yes, because `terraform state mv` should only be run on resources that need to be renamed. If `terraform state mv` fails, that's indicative of a larger problem (maybe a bug in the aforementioned check) that requires attention.

**Comment by natanlao** [comment]
@achave11 nicely volunteered to be my guinea pig. We tested it together. Here is his transcript. I've edited it for brevity.

```console
abrahamchavez at Abrahams-MacBook-Pro in ~/PycharmProjects/Azul/azul.sc on local/develop
$ make deploy  # Fully deployed
...
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.


abrahamchavez at Abrahams-MacBook-Pro in ~/PycharmProjects/Azul/azul.sc/terraform on issues/natanlao/2033-rename-resources
$ terraform state list
data.aws_caller_identity.current
data.aws_region.current
data.aws_route53_zone.azul_url
data.aws_route53_zone.dev_singlecell_gi_ucsc_edu
data.google_client_config.current
aws_acm_certificate.indexer_0
aws_acm_certificate.service_0
aws_acm_certificate_validation.indexer_0
aws_acm_certificate_validation.service_0
aws_api_gateway_base_path_mapping.indexer_0
aws_api_gateway_base_path_mapping.service_0
aws_api_gateway_deployment.indexer
aws_api_gateway_deployment.service
aws_api_gateway_domain_name.indexer_0
aws_api_gateway_domain_name.service_0
aws_dynamodb_table.cart_items_table
aws_dynamodb_table.carts_table
aws_dynamodb_table.users_table
aws_dynamodb_table.versions_table
aws_iam_role.indexer
aws_iam_role.service
aws_iam_role.state_machine_iam_role
aws_iam_role_policy.indexer
aws_iam_role_policy.service
aws_iam_role_policy.state_machine_iam_policy
aws_lambda_layer_version.dependencies
aws_route53_record.indexer_0
aws_route53_record.indexer_domain_validation_0
aws_route53_record.service_0
aws_route53_record.service_domain_validation_0
aws_route53_record.url_redirect_record
aws_s3_bucket.bucket
aws_s3_bucket.url_bucket
aws_sfn_state_machine.cart_export_state_machine
aws_sfn_state_machine.cart_item_state_machine
aws_sfn_state_machine.manifest_state_machine
aws_sqs_queue.notifications
aws_sqs_queue.notifications_fail
aws_sqs_queue.tallies
aws_sqs_queue.tallies_fail
aws_sqs_queue.tallies_retry
google_service_account.indexer
null_resource.hmac-secret
module.chalice_indexer.data.aws_caller_identity.chalice
module.chalice_indexer.data.aws_region.chalice
module.chalice_indexer.data.null_data_source.chalice
module.chalice_indexer.data.template_file.chalice_api_swagger
module.chalice_indexer.aws_api_gateway_deployment.rest_api
module.chalice_indexer.aws_api_gateway_rest_api.rest_api
module.chalice_indexer.aws_cloudwatch_event_rule.indexercachehealth-event
module.chalice_indexer.aws_cloudwatch_event_target.indexercachehealth-event
module.chalice_indexer.aws_lambda_event_source_mapping.aggregate-sqs-event-source
module.chalice_indexer.aws_lambda_event_source_mapping.aggregate_retry-sqs-event-source
module.chalice_indexer.aws_lambda_event_source_mapping.contribute-sqs-event-source
module.chalice_indexer.aws_lambda_function.aggregate
module.chalice_indexer.aws_lambda_function.aggregate_retry
module.chalice_indexer.aws_lambda_function.api_handler
module.chalice_indexer.aws_lambda_function.contribute
module.chalice_indexer.aws_lambda_function.indexercachehealth
module.chalice_indexer.aws_lambda_permission.indexercachehealth-event
module.chalice_indexer.aws_lambda_permission.rest_api_invoke
module.chalice_service.data.aws_caller_identity.chalice
module.chalice_service.data.aws_region.chalice
module.chalice_service.data.null_data_source.chalice
module.chalice_service.data.template_file.chalice_api_swagger
module.chalice_service.aws_api_gateway_deployment.rest_api
module.chalice_service.aws_api_gateway_rest_api.rest_api
module.chalice_service.aws_cloudwatch_event_rule.servicecachehealth-event
module.chalice_service.aws_cloudwatch_event_target.servicecachehealth-event
module.chalice_service.aws_lambda_function.api_handler
module.chalice_service.aws_lambda_function.cartexportpush
module.chalice_service.aws_lambda_function.cartitemwrite
module.chalice_service.aws_lambda_function.jwt_auth
module.chalice_service.aws_lambda_function.manifest
module.chalice_service.aws_lambda_function.servicecachehealth
module.chalice_service.aws_lambda_permission.jwt_auth_invoke
module.chalice_service.aws_lambda_permission.rest_api_invoke
module.chalice_service.aws_lambda_permission.servicecachehealth-event


abrahamchavez at Abrahams-MacBook-Pro in ~/PycharmProjects/Azul/azul.sc/terraform on issues/natanlao/2033-rename-resources
$ make deploy
...
python /Users/abrahamchavez/PycharmProjects/Azul/azul.sc/scripts/rename_resources.py
2020-08-07 16:55:57,618 INFO    MainThread: Found 'aws_dynamodb_table.cart_items_table', renaming to 'aws_dynamodb_table.cart_items'
2020-08-07 16:56:05,253 INFO    MainThread: Found 'aws_dynamodb_table.carts_table', renaming to 'aws_dynamodb_table.carts'
2020-08-07 16:56:12,839 INFO    MainThread: Found 'aws_dynamodb_table.users_table', renaming to 'aws_dynamodb_table.users'
2020-08-07 16:56:20,900 INFO    MainThread: Found 'aws_dynamodb_table.versions_table', renaming to 'aws_dynamodb_table.object_versions'
2020-08-07 16:56:28,478 INFO    MainThread: Found 'aws_iam_role.state_machine_iam_role', renaming to 'aws_iam_role.states'
2020-08-07 16:56:36,184 INFO    MainThread: Found 'aws_iam_role_policy.state_machine_iam_policy', renaming to 'aws_iam_role_policy.states'
2020-08-07 16:56:44,443 INFO    MainThread: Found 'aws_s3_bucket.bucket', renaming to 'aws_s3_bucket.storage'
2020-08-07 16:56:52,516 INFO    MainThread: Found 'aws_s3_bucket.url_bucket', renaming to 'aws_s3_bucket.urls'
2020-08-07 16:57:00,099 INFO    MainThread: Found 'aws_sfn_state_machine.cart_export_state_machine', renaming to 'aws_sfn_state_machine.cart_export'
2020-08-07 16:57:07,594 INFO    MainThread: Found 'aws_sfn_state_machine.cart_item_state_machine', renaming to 'aws_sfn_state_machine.cart_item'
2020-08-07 16:57:15,095 INFO    MainThread: Found 'aws_sfn_state_machine.manifest_state_machine', renaming to 'aws_sfn_state_machine.manifest'
2020-08-07 16:57:23,159 INFO    MainThread: Found 'null_resource.hmac-secret', renaming to 'null_resource.hmac_secret'
...
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.


abrahamchavez at Abrahams-MacBook-Pro in ~/PycharmProjects/Azul/azul.sc on issues/natanlao/2033-rename-resources
$ make deploy  # Nothing renamed if `make deploy` is run again
...
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

abrahamchavez at Abrahams-MacBook-Pro in ~/PycharmProjects/Azul/azul.sc/terraform on issues/natanlao/2033-rename-resources
$ terraform state list
data.aws_caller_identity.current
data.aws_region.current
data.aws_route53_zone.azul_url
data.aws_route53_zone.dev_singlecell_gi_ucsc_edu
data.google_client_config.current
aws_acm_certificate.indexer_0
aws_acm_certificate.service_0
aws_acm_certificate_validation.indexer_0
aws_acm_certificate_validation.service_0
aws_api_gateway_base_path_mapping.indexer_0
aws_api_gateway_base_path_mapping.service_0
aws_api_gateway_deployment.indexer
aws_api_gateway_deployment.service
aws_api_gateway_domain_name.indexer_0
aws_api_gateway_domain_name.service_0
aws_dynamodb_table.cart_items
aws_dynamodb_table.carts
aws_dynamodb_table.object_versions
aws_dynamodb_table.users
aws_iam_role.indexer
aws_iam_role.service
aws_iam_role.states
aws_iam_role_policy.indexer
aws_iam_role_policy.service
aws_iam_role_policy.states
aws_lambda_layer_version.dependencies
aws_route53_record.indexer_0
aws_route53_record.indexer_domain_validation_0
aws_route53_record.service_0
aws_route53_record.service_domain_validation_0
aws_route53_record.url_redirect_record
aws_s3_bucket.storage
aws_s3_bucket.urls
aws_sfn_state_machine.cart_export
aws_sfn_state_machine.cart_item
aws_sfn_state_machine.manifest
aws_sqs_queue.notifications
aws_sqs_queue.notifications_fail
aws_sqs_queue.tallies
aws_sqs_queue.tallies_fail
aws_sqs_queue.tallies_retry
google_service_account.indexer
null_resource.hmac_secret
module.chalice_indexer.data.aws_caller_identity.chalice
module.chalice_indexer.data.aws_region.chalice
module.chalice_indexer.data.null_data_source.chalice
module.chalice_indexer.data.template_file.chalice_api_swagger
module.chalice_indexer.aws_api_gateway_deployment.rest_api
module.chalice_indexer.aws_api_gateway_rest_api.rest_api
module.chalice_indexer.aws_cloudwatch_event_rule.indexercachehealth-event
module.chalice_indexer.aws_cloudwatch_event_target.indexercachehealth-event
module.chalice_indexer.aws_lambda_event_source_mapping.aggregate-sqs-event-source
module.chalice_indexer.aws_lambda_event_source_mapping.aggregate_retry-sqs-event-source
module.chalice_indexer.aws_lambda_event_source_mapping.contribute-sqs-event-source
module.chalice_indexer.aws_lambda_function.aggregate
module.chalice_indexer.aws_lambda_function.aggregate_retry
module.chalice_indexer.aws_lambda_function.api_handler
module.chalice_indexer.aws_lambda_function.contribute
module.chalice_indexer.aws_lambda_function.indexercachehealth
module.chalice_indexer.aws_lambda_permission.indexercachehealth-event
module.chalice_indexer.aws_lambda_permission.rest_api_invoke
module.chalice_service.data.aws_caller_identity.chalice
module.chalice_service.data.aws_region.chalice
module.chalice_service.data.null_data_source.chalice
module.chalice_service.data.template_file.chalice_api_swagger
module.chalice_service.aws_api_gateway_deployment.rest_api
module.chalice_service.aws_api_gateway_rest_api.rest_api
module.chalice_service.aws_cloudwatch_event_rule.servicecachehealth-event
module.chalice_service.aws_cloudwatch_event_target.servicecachehealth-event
module.chalice_service.aws_lambda_function.api_handler
module.chalice_service.aws_lambda_function.cartexportpush
module.chalice_service.aws_lambda_function.cartitemwrite
module.chalice_service.aws_lambda_function.jwt_auth
module.chalice_service.aws_lambda_function.manifest
module.chalice_service.aws_lambda_function.servicecachehealth
module.chalice_service.aws_lambda_permission.jwt_auth_invoke
module.chalice_service.aws_lambda_permission.rest_api_invoke
module.chalice_service.aws_lambda_permission.servicecachehealth-event
```

File: common.mk (hunk 1 above)
Comment (hannes-ucsc) [reply common.mk:79]: This shouldn't be in common.mk. That file is for targets that should be available in every other Makefile.

And I think @jessebrennan is right about the dependency on the init target in `terraform/Makefile`. You should try running this script after running `make clean` in that directory. You can add a convenience target in the root Makefile that calls this target in the `terraform/Makefile`. There is already precedent for that.

Comment (hannes-ucsc) [reply common.mk:79]: Now I get it. This IS mentioned again in the two terraform Makefiles. Still, I don't this should be here because it makes it available to every other Makefile and shows up in tab completion. Don't worry about Gitlab for now since you can't test it. Just define this target in the terraform/Makefile.

Comment (hannes-ucsc) [reply common.mk:79]: Now I get it. This IS mentioned again in the two terraform Makefiles. Still, I don't this should be here because it makes it available to every other Makefile and shows up in tab completion. Don't worry about Gitlab for now since you can't test it. Just define this target in the terraform/Makefile.

**Comment by natanlao** [comment]
I noticed a bug in my code:

https://github.com/DataBiosphere/azul/pull/2094/commits/e189384b431746f65b3e1d9ed73a45ca95b765c9#diff-e8b6453f0e715d783324c48feb0c4e39R39

```python
     for resource in tf_state.stdout.splitlines():
         if resource in renamed.keys():
```

`.splitlines()` generates a list of `bytes` instead of a list of `str`. Since keys of `renamed` are of type `str`, the next line (`if resource in renamed...`) will always be false.

This should have been caught when Abraham and I tested this, but it wasn't. I think we tested off a stale branch before the last set of revisions. This is fixed in the latest commit, and I'll test again with Abraham on Monday.

**Comment by hannes-ucsc** [comment]
Let's wait with the next test until I've reviewed the changes. I want to be mindful of @achave11's time.

File: scripts/rename_resources.py [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+
+def main():
+    run_args = {
+        'cwd': os.path.join(os.environ['project_root'], 'terraform'),
+        'check': True,
+        'capture_output': True,
+        'shell': False
+    }
+    tf_state = subprocess.run(['terraform', 'state', 'list'], **run_args)
+    current = [res.decode('utf-8') for res in tf_state.stdout.splitlines()]
```
Comment (hannes-ucsc) [inline scripts/rename_resources.py:39]:
```suggestion
    current = [res.decode() for res in tf_state.stdout.splitlines()]
```

utf8 is the default

File: scripts/rename_resources.py [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+    for resource in current:
+        if resource in renamed.keys():
```
Comment (hannes-ucsc) [inline scripts/rename_resources.py:41]:
```suggestion
        if resource in renamed:
```

equivalent

File: scripts/rename_resources.py [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+            log.info('Found %r, renaming to %r', resource, renamed[resource])
```
Comment (hannes-ucsc) [inline scripts/rename_resources.py:42]: https://github.com/DataBiosphere/azul/pull/1701#discussion_r456125042

File: scripts/rename_resources.py [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+    'aws_s3_bucket.url_bucket': 'aws_s3_bucket.urls',
+    'aws_sfn_state_machine.cart_export_state_machine': 'aws_sfn_state_machine.cart_export',
+    'aws_sfn_state_machine.cart_item_state_machine': 'aws_sfn_state_machine.cart_item',
+    'aws_sfn_state_machine.manifest_state_machine': 'aws_sfn_state_machine.manifest',
+    'null_resource.hmac-secret': 'null_resource.hmac_secret'
+}
+
+
+def main():
+    run_args = {
```
Comment (hannes-ucsc) [inline scripts/rename_resources.py:32]: If you defined a function called `terraform_state()` you might be able to reduce this further.

File: scripts/rename_resources.py [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+    }
+    tf_state = subprocess.run(['terraform', 'state', 'list'], **run_args)
+    current = [res.decode('utf-8') for res in tf_state.stdout.splitlines()]
+    for resource in current:
+        if resource in renamed.keys():
+            log.info('Found %r, renaming to %r', resource, renamed[resource])
+            subprocess.run(['terraform', 'state', 'mv', resource, renamed[resource]],
+                           **run_args)
+        else:
+            log.debug('Found %r, does not need to be renamed', resource)
```
Comment (hannes-ucsc) [inline scripts/rename_resources.py:46]: This logs the same message for resources that aren't mentioned in the mapping and resources that are already at the new name. I'd prefer if it logged nothing in the first case. And

```suggestion
            log.info('Found %r, does not need to be renamed', resource)
```

File: scripts/rename_resources.py (hunk 11 above)
Comment (hannes-ucsc) [reply scripts/rename_resources.py:39]: And its probably faster to decode the entire output first and then split.

File: scripts/rename_resources.py (hunk 13 above)
Comment (natanlao) [reply scripts/rename_resources.py:42]: Ah, my bad. Sorry for getting this wrong so many times.

File: scripts/rename_resources.py [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+                          capture_output=True,
+                          shell=False)
+
+
+def main():
+    for resource in terraform_state('list').stdout.decode().splitlines():
+        try:
+            log.info('Found %r, renaming to %r', resource, renamed[resource])
+        except KeyError:
+            continue
```
[the remark this answers is not in the document]
Comment (hannes-ucsc) [reply scripts/rename_resources.py:44]: In https://github.com/DataBiosphere/azul/pull/2094#discussion_r467495498 I was referring to two cases. One should log, the other one shouldn't.

File: scripts/rename_resources.py [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+                          check=True,
+                          capture_output=True,
+                          shell=False)
+
+
+def main():
+    current_names = terraform_state('list').stdout.decode().splitlines()
+    for current_name in current_names:
+        try:
+            new_name = renamed[current_name]
```
Comment (hannes-ucsc) [inline scripts/rename_resources.py:44]: minimized try block, mentioned in C guide

File: scripts/rename_resources.py [hunk 18]
```diff
@@ -0,0 +1,55 @@
...
+        except KeyError:
+            if current_name in renamed.values():
+                log.info('Found %r, already renamed', current_name)
```
Comment (hannes-ucsc) [inline scripts/rename_resources.py:47]: `else: pass` served no purpose

File: scripts/rename_resources.py [hunk 19]
```diff
@@ -0,0 +1,55 @@
...
+def terraform_state(*args):
+    return subprocess.run(['terraform', 'state', *args],
+                          cwd=os.path.join(os.environ['project_root'], 'terraform'),
+                          check=True,
+                          capture_output=True,
+                          shell=False)
+
+
+def main():
+    current_names = terraform_state('list').stdout.decode().splitlines()
```
Comment (hannes-ucsc) [inline scripts/rename_resources.py:41]: I think it's a little more readable with those new names.

**Comment by hannes-ucsc** [comment]
Alright, we need one more guinea pig to test this.

**Comment by natanlao** [comment]
Tested again -- thanks, @achave11. Here's an abridged transcript. We confirmed we were working off the latest `issues/natanlao/2033-rename-resources` before testing.

```console
$ make deploy
...
python /Users/abrahamchavez/PycharmProjects/Azul/azul.sc/scripts/rename_resources.py
2020-08-13 16:11:28,927 INFO    MainThread: Found 'aws_dynamodb_table.cart_items_table', renaming to 'aws_dynamodb_table.cart_items'
2020-08-13 16:11:37,138 INFO    MainThread: Found 'aws_dynamodb_table.carts_table', renaming to 'aws_dynamodb_table.carts'
2020-08-13 16:11:45,381 INFO    MainThread: Found 'aws_dynamodb_table.users_table', renaming to 'aws_dynamodb_table.users'
2020-08-13 16:11:54,498 INFO    MainThread: Found 'aws_dynamodb_table.versions_table', renaming to 'aws_dynamodb_table.object_versions'
2020-08-13 16:12:02,286 INFO    MainThread: Found 'aws_iam_role.state_machine_iam_role', renaming to 'aws_iam_role.states'
2020-08-13 16:12:10,055 INFO    MainThread: Found 'aws_iam_role_policy.state_machine_iam_policy', renaming to 'aws_iam_role_policy.states'
2020-08-13 16:12:18,186 INFO    MainThread: Found 'aws_s3_bucket.bucket', renaming to 'aws_s3_bucket.storage'
2020-08-13 16:12:26,081 INFO    MainThread: Found 'aws_s3_bucket.url_bucket', renaming to 'aws_s3_bucket.urls'
2020-08-13 16:12:34,848 INFO    MainThread: Found 'aws_sfn_state_machine.cart_export_state_machine', renaming to 'aws_sfn_state_machine.cart_export'
2020-08-13 16:12:43,178 INFO    MainThread: Found 'aws_sfn_state_machine.cart_item_state_machine', renaming to 'aws_sfn_state_machine.cart_item'
2020-08-13 16:12:52,078 INFO    MainThread: Found 'aws_sfn_state_machine.manifest_state_machine', renaming to 'aws_sfn_state_machine.manifest'
2020-08-13 16:12:59,721 INFO    MainThread: Found 'null_resource.hmac-secret', renaming to 'null_resource.hmac_secret'
...

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
...
$ make deploy
...
python /Users/abrahamchavez/PycharmProjects/Azul/azul.sc/scripts/rename_resources.py
2020-08-13 16:21:38,583 INFO    MainThread: Found 'aws_dynamodb_table.cart_items', already renamed
2020-08-13 16:21:38,583 INFO    MainThread: Found 'aws_dynamodb_table.carts', already renamed
2020-08-13 16:21:38,583 INFO    MainThread: Found 'aws_dynamodb_table.object_versions', already renamed
2020-08-13 16:21:38,583 INFO    MainThread: Found 'aws_dynamodb_table.users', already renamed
2020-08-13 16:21:38,583 INFO    MainThread: Found 'aws_iam_role.states', already renamed
2020-08-13 16:21:38,583 INFO    MainThread: Found 'aws_iam_role_policy.states', already renamed
2020-08-13 16:21:38,583 INFO    MainThread: Found 'aws_s3_bucket.storage', already renamed
2020-08-13 16:21:38,584 INFO    MainThread: Found 'aws_s3_bucket.urls', already renamed
2020-08-13 16:21:38,584 INFO    MainThread: Found 'aws_sfn_state_machine.cart_export', already renamed
2020-08-13 16:21:38,584 INFO    MainThread: Found 'aws_sfn_state_machine.cart_item', already renamed
2020-08-13 16:21:38,584 INFO    MainThread: Found 'aws_sfn_state_machine.manifest', already renamed
2020-08-13 16:21:38,584 INFO    MainThread: Found 'null_resource.hmac_secret', already renamed
...

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
...
```

**Comment by natanlao** [comment]
Progress and revisions on this PR from standup discussion with @hannes-ucsc:
- `config.qualified_resource_name` will choke on `aws_cloudwatch_event_rule`s generated by Chalice (e.g. `servicecachehealth-event`). Solution is to drop `-event` suffix on all such resources (e.g. `servicecachehealth`) in Terraform config post-processing that occurs in `scripts/prepare_lambda_deployment.py` using a two-pass algorithm that (first) catalogs and renames resource names and (second) makes changes to references to resource names. As this is chained on the resource renaming PR, we can add to that script in this PR to rename already-deployed resources with those names.
- Script to generate list of taggable resources can be moved into azul.terraform directly so that it is run on each deploy. If the delay is annoying (between 5-10s) then we can address that at a later date.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
This is marked as `chain` but has no dependencies (blocked PR).

The comment https://github.com/DataBiosphere/azul/pull/2094#issuecomment-677894988 sounds like it applies more to the chained PR than this one,

**Comment by natanlao** [comment]
> This is marked as chain but has no dependency (blocked PR).

I accidentally blocked the issue and not the PR. I've fixed this.

**Review by hannes-ucsc — APPROVED** [review]

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/jobs/5656

**Comment by natanlao** [comment]
This failed test is interesting; the cycle should have been caught when `terraform plan` was run (hashicorp/terraform#22561). It looks like this error is occurring because the `rename_resources` target is a dependency of only the `apply` target and not the `auto_apply` target.

I'm testing a fix now.

**Review by hannes-ucsc — APPROVED** [review]
